chain-alerter/src/alerts.rs

```diff
@@ -1,17 +1,113 @@
 //! Specific chain alerts.
 
-use crate::format::fmt_amount;
+use crate::format::{fmt_amount, fmt_duration};
 use crate::slack::SlackClientInfo;
-use crate::subspace::{AI3, BlockInfo, ExtrinsicInfo, SubspaceConfig};
+use crate::subspace::{
+    AI3, BlockInfo, ExtrinsicInfo, SubspaceConfig, gap_since_last_block, gap_since_time,
+};
+use chrono::Utc;
 use scale_value::Composite;
+use std::sync::Arc;
 use std::time::Duration;
 use subxt::blocks::ExtrinsicDetails;
 use subxt::client::OnlineClientT;
+use tokio::sync::watch;
+use tokio::time::sleep;
 use tracing::warn;
 
 /// The minimum balance change to alert on.
 const MIN_BALANCE_CHANGE: u128 = 1_000_000_000 * AI3;
 
+/// The minimum gap between block timestamps to alert on.
+/// The target block gap is 6 seconds, so we alert if it takes substantially longer.
+///
+/// `pallet-timestamp` enforces a `MinimumPeriod` of 3 seconds in Subspace, and a
+/// `MAX_TIMESTAMP_DRIFT_MILLIS` of 30 seconds from each node's local clock.
+/// <https://github.com/paritytech/polkadot-sdk/blob/0034d178fff88a0fd87cf0ec1d8f122ae0011d78/substrate/frame/timestamp/src/lib.rs#L307>
+const MIN_BLOCK_GAP: Duration = Duration::from_secs(60);
```

Comment:
@jfrank-summit what block time gap should we alert on?

Currently we check both:
- the difference between the local time and the block time (if at least this gap has elapsed locally)
- the difference between each pair of block times

The largest block gap I've seen in testing is 25 seconds.

Reply:
60 seconds is good for now.